lib/src/summarize.dart

```diff
@@ -4,12 +4,95 @@
 
 library dart_pad.summarize;
 
-/// Instances of this class take string input of dart code and output a text 
-/// description ofthe code's size, packages, and other useful information
+import '../services/dartservices.dart';
+/// Instances of this class take string input of dart code as well as an analysis result,
+/// and output a text description ofthe code's size, packages, and other useful information.
 class Summarizer {
   
-  String summarize(String input) {
-    return "foo";
+  SummarizeToken storage;
+
+  bool resultsPresent;
+  
+  Summarizer(String input, [AnalysisResults analysis]) {
+    if (input == null) throw new ArgumentError("Input can't be null.");
+    resultsPresent = !(analysis == null);
+    storage = new SummarizeToken(input, analysis);
   }
+  
+  String returnAsSimpleSummary() {
+    String summary = 'Summary (Under Development):';
+    return summary;
+  }
+  
+  String returnAsGistMarkDown() {
+    if (resultsPresent) {
+      String summary = '<pre><code><b>Summary (Under Development)</b><p/>';
+      summary +='${storage.linesCode} lines of code. <p/>';
+      if (storage.errorPresent) summary += 'Errors are present.<p/>';
+      if (storage.errorPresent) summary += 'Warnings are present.<p/>';
+      summary += '${storage.features} <p/>';
+      summary += '${storage.packageCount} Package Imports: ${storage.packageImports}. <p/>';
+      summary += '${storage.resolvedCount} Resolved Imports: ${storage.resolvedImports}. <p/>';
+      summary += 'List of ${storage.errorCount} Errors: <ul>';
+      for (AnalysisIssue issue in storage.errors) {
+        summary += '<li>${_condenseIssue(issue)}</li>';
+      }
+      summary += '</ul></code></pre>';
+      return summary;
+    }
+    else {
+      String summary = '<pre><code><b>Summary (Under Development)</b><p/>';
+      summary +='${storage.linesCode} lines of code. <p/>';
+      return summary;
+    }
+    
+  }
+  
+  String _condenseIssue(AnalysisIssue issue) {
+    return '''${issue.kind.toUpperCase()} | ${issue.message} </n>
+  Source at ${issue.sourceName} <n/>
+  Located at line: ${issue.line} <n/>.
+  ''';
+    }
 }
 
+class SummarizeToken {
```

Comment:
Let's make this class private, as people outside the library don't need to know about it (`_ SummarizeToken`).